### PR TITLE
Create project templates target to 10.0

### PR DESCRIPTION
**Pull request by v-wuzhai**

### CODE REVIEW

**Comment by v-wuzhai** [comment]
Looks like we need to merge the #43015 code flow first to address the error message.

`error NETSDK1045: The current .NET SDK does not support targeting .NET 10.0.  Either target .NET 9.0 or lower, or use a version of the .NET SDK that supports .NET 10.0. Download the .NET SDK from https://aka.ms/dotnet/download`

**Comment by marcpopMSFT** [comment]
@v-wuzhai the other PR is now in so we can fix the merge conflicts and start getting this one unblocked.

File: test/dotnet-new.Tests/Approvals/DotnetNewHelpTests.CanShowHelpForTemplate_MatchOnLanguage.verified.txt [hunk 1]
```diff
@@ -16,16 +16,16 @@ Options:
...
-                                                 Type: choice
-                                                   net9.0  Target net9.0
-                                                   net8.0  Target net8.0
-                                                   net7.0  Target net7.0
-                                                   net6.0  Target net6.0
-                                                 Default: net9.0
-  --no-restore                                   If specified, skips the automatic restore of the project on create.
-                                                 Type: bool
-                                                 Default: false
+  -f, --framework <net10.0|net6.0|net7.0|net8.0>  The target framework for the project.
```
Comment (marcpopMSFT) [inline test/dotnet-new.Tests/Approvals/DotnetNewHelpTests.CanShowHelpForTemplate_MatchOnLanguage.verified.txt:19]: Why is net9 missing from this list? Shouldn't that show up if 9.0 is installed? Is that missing somewhere from the test setup (I see it in the restoretoolset scripts so not sure)?

**Comment by marcpopMSFT** [comment]
Should we remove the 9.0 templates (like we removed 8.0 from the 9.0 branch)?
Should we unwind https://github.com/dotnet/sdk/pull/42743/commits/426b0434df05a5a3fbbaa550639e7f2147ba32b4?

We can do those changes in a separate PR. Still not sure why the net9.0 version is missing from the test run as it'll be needed eventually. Is that because we're using a VS image that doesn't have 9 yet maybe?

File: test/dotnet-new.Tests/Approvals/DotnetNewHelpTests.CanShowHelpForTemplate_MatchOnLanguage.verified.txt (hunk 1 above)
Comment (marcpopMSFT) [reply test/dotnet-new.Tests/Approvals/DotnetNewHelpTests.CanShowHelpForTemplate_MatchOnLanguage.verified.txt:19]: tested locally, we only show the net version when those templates are installed (presumably by another SDK or by VS). I think the build machines we're building with are still 17.11 versions so no .net9 templates installed yet so this is ok for now but will break once the VS images get updated.

**Review by marcpopMSFT — APPROVED** [review]
Approving assuming we'll do the other changes I outlined in a separate PR.

**Comment by v-wuzhai** [comment]
@marcpopMSFT I will submit a separate PR to address the issues you mentioned.

**Comment by v-wuzhai** [comment]
@marcpopMSFT I have created an issue to track the missing net9.0 templates: dotnet/sdk#44736. We will update our test environment once the templates are available.

**Pull request merged**